packages/admin/admin/src/common/buttons/feedback/FeedbackButton.tsx

```diff
@@ -31,24 +31,41 @@ const Tooltip = createComponentSlot(CometTooltip)<FeedbackButtonClassKey>({
     slotName: "tooltip",
 })();
 
-export interface FeedbackButtonProps
+type FeedbackStateControl = "props" | "onClickPromise";
+
+export interface FeedbackButtonPropsBase
     extends ThemedComponentBaseProps<{
             root: typeof LoadingButton;
             tooltip: typeof CometTooltip;
         }>,
-        LoadingButtonProps {
-    loading?: boolean;
-    hasErrors?: boolean;
+        Omit<LoadingButtonProps, "loading"> {
+    feedbackStateControl: FeedbackStateControl;
     startIcon?: React.ReactNode;
     endIcon?: React.ReactNode;
     tooltipSuccessMessage?: React.ReactNode;
     tooltipErrorMessage?: React.ReactNode;
 }
 
+export interface FeedbackButtonProps extends FeedbackButtonPropsBase {
+    feedbackStateControl: "props";
+    onClick?: () => void;
+    hasErrors?: boolean;
+    loading?: boolean;
+}
+
+export interface FeedbackButtonPropsPromise extends FeedbackButtonPropsBase {
+    feedbackStateControl: "onClickPromise";
```

Comment:
Couldn't this be inferred based on the passed onClick handler? Is it possible to detect if a function is async or not?

Reply:
There are ways but it's not straightforward and probably also not reliable. Additionally, it is way more descriptive with the flag. So users will actually understand whats going on.

Reply:
@nsams @thomasdax98 what do you think?

Reply:
I would prefer a solution that doesn't need this feedbackStateControl

Reply:
Okay. The other two options I can think of are:
1. Somehow infer from function if it is async or not
2. Add a second `onClickAsync` prop for the promise-based usage

Reply:
1. sounds good.

`if (onClick instanceof Promise)`

Reply:
That doesn't work like this unfortunately. You would have to execute the function and then do this check.
In this thread are a few options. Imo nothing is really nice^^
https://stackoverflow.com/questions/38508420/how-to-know-if-a-function-is-async

Reply:
Wouldn't something like this work?

```ts
onClick={() => {
  const onClickResult = onClick();
  const isPromise = onClickResult !== undefined;

  if(isPromise) {
    await onClickResult;
  } else {
    ...
  }
}
```

Or is this too late?

Reply:
That's too late unfortunately. I could infer that a user wants to rely on outside state if loading and/or hasError is not undefined 🤷🏻‍♂️

Reply:
> That's too late unfortunately. I could infer that a user wants to rely on outside state if loading and/or hasError is not undefined 🤷🏻‍♂️

I like this idea! Could you please try this? Also, I'd still recommend adding the promise detection in a separate PR: https://github.com/vivid-planet/comet/pull/2237#pullrequestreview-2211777135

Reply:
Here to following PR: https://github.com/vivid-planet/comet/pull/2425

Reply:
I will close this one